config/confighttp/confighttp.go

```diff
@@ -227,9 +232,18 @@ func (hcs *ClientConfig) ToClient(ctx context.Context, host component.Host, sett
 		}
 	}
 
+	var jar http.CookieJar
+	if hcs.CookiesEnabled {
+		jar, err = cookiejar.New(nil)
+		if err != nil {
+			return nil, err
```

Comment:
This line cannot be tested, the New method returns nil always for now. I am just checking the error returned in case this changes in the future.